src/StoreApi/Authentication.php

```diff
@@ -55,4 +93,18 @@ protected function is_request_to_store_api() {
 		}
 		return 0 === strpos( $GLOBALS['wp']->query_vars['rest_route'], '/wc/store/' );
 	}
+
+	/**
+	 * Get current user IP Address.
+	 *
+	 * Note for HTTP_X_FORWARDED_FOR, Proxy servers can send through this header like this: X-Forwarded-For: client1, proxy1, proxy2.
+	 * Make sure we always only send through the first IP in the list which should always be the client IP.
+	 *
+	 * @return string
+	 */
+	protected static function get_ip_address() {
+		$ip_address = trim( current( explode( ',', sanitize_text_field( wp_unslash( $_SERVER['HTTP_X_REAL_IP'] ?? $_SERVER['HTTP_X_FORWARDED_FOR'] ?? $_SERVER['REMOTE_ADDR'] ?? '' ) ) ) ) );
```

Comment:
We might bump into some problems here and break functionality for some users. `$_SERVER['HTTP_X_REAL_IP']` and `$_SERVER['HTTP_X_FORWARDED_FOR']` can very easily get spoofed as they mirror the respective request headers, and `$_SERVER['REMOTE_ADDR']` will report wrongly if behind some proxy, or some caching system like Varnish or Cloudflare. This means a real attacker will bypass this very easily, and a real user might get blocked easily too.
`$_SERVER['REMOTE_ADDR']` can also be spoofed if the attacker would just want to abuse a certain endpoint, not caring about getting a response.

Also, I am not sure returning a default value (empty string here) would be the way to go, as it would group all users in that condition (well they were up to no good probably). I wonder if it wouldn't make sense to error out the request.

All this being said, one actionable thing we can do, is to expand this function, remove the confusing one-liner and increase readability as there is a fair chance someone has to tweak this in the future.

One more thing, do we have data on how many users resort to Cloudflare? Because if I remember correctly, Cloudflare passes the proxies through some specific headers, if it justifies, we should address that case here too.